web/src/main/java/org/cbioportal/web/util/StudyViewFilterApplier.java

```diff
@@ -142,25 +148,70 @@ public SampleIdentifier apply(Sample sample) {
         if (mutationCountVsCNASelection != null && !sampleIdentifiers.isEmpty()) {
             sampleIdentifiers = filterMutationCountVsCNASelection(mutationCountVsCNASelection, sampleIdentifiers);
         }
+        
+        Boolean withOncoKBDriverMutationData = studyViewFilter.getwithOncoKBDriverMutationData();
+        if (withOncoKBDriverMutationData != null && !sampleIdentifiers.isEmpty()) {
+            sampleIdentifiers = filterByDriverMutation(sampleIdentifiers, withOncoKBDriverMutationData);
+        }
+        
+        // Filter the samples if the withOncoKBDriverMutationData is specified.
+        // Please see withMutationData as example above
+        // Please use the function filterByDriverMutation defined bellow
+        
 
         return sampleIdentifiers;
     }
     
+    
+    private List<SampleIdentifier> filterByDriverMutation(List<SampleIdentifier> sampleIdentifiers, Boolean withOncoKBDriverMutationData) {
+        // Based on withOncoKBDriverMutationData, you need to convert that value to oncogenicity which the OncoKB
+        // This method should be used: oncoKBConverter.getOncogenicityByHasDriver()
+        // After getting oncogenicities, you need to construct the List<AnnotationFilter> for the function below
+        // The method need to used to get all qualified mutations: mutationService.getMutationsInMultipleMolecularProfilesByAnnotation()
+        // After getting the mutations, just get the list of samples from the mutations and return
+        List<String> studyIds = new ArrayList<>();
+        List<String> sampleIds = new ArrayList<>();
+        studyViewFilterUtil.extractStudyAndSampleIds(sampleIdentifiers, studyIds, sampleIds);
+        
+        List<String> oncogenicityIdentifier;
+        if(withOncoKBDriverMutationData == true) {
+          oncogenicityIdentifier = oncoKBConverter.getOncogenicityByHasDriver("YES");
+        }
+        if(withOncoKBDriverMutationData == false) {
+          oncogenicityIdentifier = oncoKBConverter.getOncogenicityByHasDriver("NO");
+        }
+        List<AnnotationFilter> annotationProfiles;
+        AnnotationFilter annotationOncoKBData = new AnnotationFilter();
+        annotationOncoKBData.setPath("oncokb.oncogenicity");
+        annotationOncoKBData.setValues(new List<String>("YES", "NO", "YES,NO"));
```

Comment:
You are doing filtering, imaging you need to get a list of mutations which qualifies the oncogenicity values we are trying to filter. If `withOncoKBDriverMutationData ` is true, `oncoKBConverter.getOncogenicityByHasDriver` will return you ['oncogenic', 'likely oncogenic', 'predicted oncogenic']. The actual mutation oncogenicity should be one of these values. Take a look at the MutationMapper.xml, how the annotationProfiles is actually used.